src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeFinaliserServiceTest.java

```diff
@@ -81,14 +81,98 @@ public void finaliseEnvelope_should_update_envelope_status_and_clear_all_ocr_dat
+        );
+
+        UUID envelopeId = envelopeRepository.saveAndFlush(envelope).getId();
+
+        // when
+        envelopeFinaliserService.finaliseEnvelope(envelopeId);
+
+        // then
+        Optional<Envelope> finalisedEnvelope = envelopeRepository.findById(envelopeId);
+
+        assertThat(finalisedEnvelope).isPresent();
+        assertThat(finalisedEnvelope.get().getStatus()).isEqualTo(Status.COMPLETED);
+        assertThat(finalisedEnvelope.get().getScannableItems())
+            .allMatch(item -> item.getOcrData() == null && item.getOcrValidationWarnings() == null);
+    }
+
+    @Test
+    public void finaliseEnvelope_should_update_envelope_status_and_clear_all_ocr_data_and_warnings() {
+        // given
+        Envelope envelope = envelope(
+            "JURISDICTION1",
+            Status.NOTIFICATION_SENT,
+            createScannableItems(3, createOcrData(), createWarnings())
+        );
+
+        UUID envelopeId = envelopeRepository.saveAndFlush(envelope).getId();
+
+        // when
+        envelopeFinaliserService.finaliseEnvelope(envelopeId);
+
+        // then
+        Optional<Envelope> finalisedEnvelope = envelopeRepository.findById(envelopeId);
+
+        assertThat(finalisedEnvelope).isPresent();
+        assertThat(finalisedEnvelope.get().getStatus()).isEqualTo(Status.COMPLETED);
+        assertThat(finalisedEnvelope.get().getScannableItems())
+            .allMatch(item -> item.getOcrData() == null && item.getOcrValidationWarnings() == null);
+    }
+
+    @Test
+    public void finaliseEnvelope_should_not_update_other_envelopes_with_ocr_data() {
+        // given
+        Envelope envelope1 = envelope(
+            "JURISDICTION1",
+            Status.NOTIFICATION_SENT,
+            createScannableItems(3, createOcrData(), null)
+        );
+        Envelope envelope2 = envelope(
+            "JURISDICTION1",
+            Status.NOTIFICATION_SENT,
+            createScannableItems(3, createOcrData(), null)
+        );
+
+        UUID envelope1Id = envelopeRepository.saveAndFlush(envelope1).getId();
+        UUID envelope2Id = envelopeRepository.saveAndFlush(envelope2).getId();
+
+        // when
+        envelopeFinaliserService.finaliseEnvelope(envelope1Id);
+
+        // then
+        Optional<Envelope> unaffectedEnvelope = envelopeRepository.findById(envelope2Id);
+
+        assertThat(unaffectedEnvelope).isPresent();
+        assertThat(unaffectedEnvelope.get().getStatus()).isEqualTo(Status.NOTIFICATION_SENT);
+        assertThat(unaffectedEnvelope.get().getScannableItems())
+            .allMatch(item -> item.getOcrData() != null && item.getOcrValidationWarnings() == null);
+    }
+
+    @Test
+    public void finaliseEnvelope_should_not_update_other_envelopes_with_warnings() {
+        // given
+        Envelope envelope1 = envelope(
+            "JURISDICTION1",
+            Status.NOTIFICATION_SENT,
+            createScannableItems(3, null, createWarnings())
```

Comment:
this its invalid scenario you can't have warnings without OCR

Reply:
Thank you, fixed this.